pkg/container-disk/container-disk.go

```diff
@@ -442,3 +441,63 @@ func toContainerName(volumeName string) string {
 func toVolumeName(containerName string) string {
 	return strings.TrimPrefix(containerName, "volume")
 }
+
+// getMinimalInitContainerDiskResources calculates resource requirements for container disk containers
+// This function is shared between init containers and regular containers
+func getMinimalInitContainerDiskResources(vmi *v1.VirtualMachineInstance, config *virtconfig.ClusterConfig) kubev1.ResourceRequirements {
+	resources := kubev1.ResourceRequirements{}
+	resources.Requests = make(kubev1.ResourceList)
+	resources.Limits = make(kubev1.ResourceList)
+
+	resources.Requests[kubev1.ResourceCPU] = resource.MustParse("1m")
+	if cpuRequest := config.GetSupportContainerRequest(v1.ContainerDisk, kubev1.ResourceCPU); cpuRequest != nil {
+		resources.Requests[kubev1.ResourceCPU] = *cpuRequest
+	}
+	resources.Requests[kubev1.ResourceMemory] = resource.MustParse("1M")
+	if memRequest := config.GetSupportContainerRequest(v1.ContainerDisk, kubev1.ResourceMemory); memRequest != nil {
+		resources.Requests[kubev1.ResourceMemory] = *memRequest
+	}
+	resources.Requests[kubev1.ResourceEphemeralStorage] = resource.MustParse(ephemeralStorageOverheadSize)
+
+	resources.Limits[kubev1.ResourceCPU] = resource.MustParse("10m")
+	if cpuLimit := config.GetSupportContainerLimit(v1.ContainerDisk, kubev1.ResourceCPU); cpuLimit != nil {
+		resources.Limits[kubev1.ResourceCPU] = *cpuLimit
+	}
+	resources.Limits[kubev1.ResourceMemory] = resource.MustParse("40M")
+	if memLimit := config.GetSupportContainerLimit(v1.ContainerDisk, kubev1.ResourceMemory); memLimit != nil {
+		resources.Limits[kubev1.ResourceMemory] = *memLimit
+	}
+
+	if vmi.IsCPUDedicated() || vmi.WantsToHaveQOSGuaranteed() {
+		resources.Requests[kubev1.ResourceCPU] = resources.Limits[kubev1.ResourceCPU]
+		resources.Requests[kubev1.ResourceMemory] = resources.Limits[kubev1.ResourceMemory]
+	}
+
+	return resources
+}
+
+// CreateImageVolumeInitContainer creates a single init container for ImageVolume feature
+func CreateImageVolumeInitContainer(vmi *v1.VirtualMachineInstance, config *virtconfig.ClusterConfig, name, image string, imagePullPolicy kubev1.PullPolicy) kubev1.Container {
+	resources := getMinimalInitContainerDiskResources(vmi, config)
+	return kubev1.Container{
+		Name:            fmt.Sprintf("volume%s", name),
+		Image:           image,
+		ImagePullPolicy: imagePullPolicy,
+		Command:         []string{filepath.Join(util.ContainerBinary, "/usr/bin/container-disk")},
+		Args:            []string{"--no-op"},
+		Resources:       resources,
+		SecurityContext: &kubev1.SecurityContext{
+			RunAsUser:                pointer.P(int64(util.NonRootUID)),
+			RunAsNonRoot:             pointer.P(true),
+			AllowPrivilegeEscalation: pointer.P(false),
+			Capabilities: &kubev1.Capabilities{
+				Drop: []kubev1.Capability{"ALL"},
+			},
+		},
+		VolumeMounts: []kubev1.VolumeMount{{
+			Name:      LauncherVolume,
+			MountPath: util.ContainerBinary,
+			ReadOnly:  true,
```

Comment:
I think using `SubPath` to only mount the relevant binary would be cool.
If this will block the PR we can wait for a follow-up to do that.